mhkit/wave/resource.py

```diff
@@ -322,87 +326,84 @@ def surface_elevation(
             raise ValueError(
-                "shape of frequency_bins must match shape of the frequency dimension of S"
+                "shape of frequency_bins must only contain 1 column and match the shape of the frequency dimension of S"
             )
+        delta_f = frequency_bins
+        delta_f_even = np.allclose(frequency_bins, frequency_bins[0])
+        if delta_f_even:
+            # reduce delta_f to a scalar if it is uniform
+            delta_f = delta_f[0].item()
+    else:
+        delta_f = f.values[1] - f.values[0]
+        delta_f_even = np.allclose(f.diff(dim=frequency_dimension)[1:], delta_f)
     if phases is not None:
-        if not list(phases.data_vars) == list(S.data_vars):
-            raise ValueError("phases must have the same variable names as S")
-        for var in phases.data_vars:
-            if not phases[var].shape == S[var].shape:
-                raise ValueError(
-                    "shape of variables in phases must match shape of variables in S"
-                )
-    if method is not None:
-        if not (method == "ifft" or method == "sum_of_sines"):
-            raise ValueError(f"Method must be 'ifft' or 'sum_of_sines'. Got: {method}")
-
+        if not phases.shape == S.shape:
+            raise ValueError(
+                "shape of variables in phases must match shape of variables in S"
+            )
     if method == "ifft":
+        # ifft method must have a zero frequency and evenly spaced frequency bins
         if not f[0] == 0:
             warnings.warn(
                 f"ifft method must have zero frequency defined. Lowest frequency is: {f[0].values}. Setting method to less efficient `sum_of_sines` method."
             )
             method = "sum_of_sines"
-
-    if frequency_bins is None:
-        delta_f = f.values[1] - f.values[0]
-        if not np.allclose(f.diff(dim=frequency_dimension)[1:], delta_f):
-            raise ValueError(
-                "Frequency bins are not evenly spaced. "
-                + "Define 'frequency_bins' or create a constant "
-                + "frequency spacing for S."
+        if not delta_f_even:
+            warnings.warn(
+                f"ifft method must have evenly spaced frequency bins. Setting method to less efficient `sum_of_sines` method."
             )
+            method = "sum_of_sines"
+    elif method == "sum_of_sines":
+        # For sum of sines, does not matter if there is a zero frequency or if frequency bins are evenly spaced
+        pass
     else:
-        if not len(frequency_bins.squeeze().shape) == 1:
-            raise ValueError("frequency_bins must only contain 1 column")
-        delta_f = frequency_bins
-        method = "sum_of_sines"
+        raise ValueError(f"Method must be 'ifft' or 'sum_of_sines'. Got: {method}")
 
     omega = xr.DataArray(
         data=2 * np.pi * f, dims=frequency_dimension, coords={frequency_dimension: f}
     )
 
-    eta = xr.Dataset()
-    for var in S.data_vars:
```

Comment:
Same as above removed the ability to iterate of multiple columns, but we still accept DataSets and multi column pandas